src/faction_camp.cpp

```diff
@@ -4637,16 +4637,18 @@ mass_volume om_harvest_itm( const npc_ptr &comp, const tripoint_abs_omt &omt_tgt
     tripoint mapmax = tripoint( 2 * SEEX - 1, 2 * SEEY - 1, omt_tgt.z() );
     for( const tripoint &p : target_bay.points_in_rectangle( mapmin, mapmax ) ) {
         for( const item &i : target_bay.i_at( p ) ) {
-            total_m += i.weight( true );
-            total_v += i.volume( true );
-            total_num += 1;
-            if( take && x_in_y( chance, 100 ) ) {
-                if( comp ) {
-                    comp->companion_mission_inv.push_back( i );
+            if( i.typeId() != itype_id( "water" ) && !i.made_of( phase_id::LIQUID ) ) {
```

Comment:
You might want to consider `item::made_of_from_type` instead of specifically checking for water. That checks for the phase from the item definition independent of the current state of the item.

Reply:
Thanks for the quick comment. "!i.made_of_from_type(phase_id::LIQUID)" is what I'm about to test, and the reason I put this into draft. Your comment is a great indicator this is a promising approach.